internal/controller/prefectserver_controller.go

```diff
@@ -270,20 +267,26 @@ func (r *PrefectServerReconciler) reconcileDeployment(ctx context.Context, serve
 
 		condition = conditions.Updated(objName)
 	} else {
-		imageVersion := prefectiov1.VersionFromImage(desiredDeployment.Spec.Template.Spec.Containers[0].Image)
-		if server.Status.Version != imageVersion ||
-			!meta.IsStatusConditionTrue(server.Status.Conditions, "DeploymentReconciled") {
+		if !meta.IsStatusConditionTrue(server.Status.Conditions, "DeploymentReconciled") {
+			condition = conditions.Updated(objName)
+		}
 
-			server.Status.Version = imageVersion
+		ready := foundDeployment.Status.AvailableReplicas > 0
+		version := prefectiov1.VersionFromImage(desiredDeployment.Spec.Template.Spec.Containers[0].Image)
 
-			condition = conditions.Updated(objName)
+		if server.Status.Ready != ready || server.Status.Version != version {
+			server.Status.Ready = ready
+			server.Status.Version = version
+
+			if statusErr := r.Status().Update(ctx, server); statusErr != nil {
+				return &ctrl.Result{}, statusErr
+			}
```

Comment:
I decided to uncouple the condition update from the update of any other status fields, because it was hard to determine if the status needed changing separately from the conditions.  This split makes it much more explicitly when we're updating the status.